src/compiler/transformers/classFields.ts

```diff
@@ -1287,6 +1310,10 @@ namespace ts {
                 Debug.assertNever(node, "Unknown class element type.");
             }
 
+            if (potentialDuplicatePrivateName) {
+                return new Error("Duplicate private name");
+            }
+

```

Comment:
Do we do anything with this message? Why not just return true/false. Maybe rename the method to tryAddPrivateIdentifierToEnvironment to signal the fact that it may fail.